src/HieDb/Utils.hs

```diff
@@ -62,16 +68,20 @@ addTypeRef (getConn -> conn) hf arr ixs sp = go 0
     el = srcSpanEndLine sp
     ec = srcSpanEndCol sp
     go :: Int -> TypeIndex -> TypeIndexing ()
-    go d i = do
+    go depth i = do
       case ixs A.! i of
         Nothing -> pure ()
         Just occ -> do
-          let ref = TypeRef occ hf d sl sc el ec
+          let ref = TypeRef occ hf depth sl sc el ec
           indexed <- get
-          when (Set.notMember (occ, d) indexed) $ do
+          let isTypeIndexed = ISet.member (fromIntegral occ) (IMap.findWithDefault ISet.empty depth indexed)
```

Comment:
I think you can use `i` to ensure this still works on 32 bit machines.

Reply:
Sorry, I don't follow.
I think we have to use the `occ` (which corresponds to type's id looked up from typenames table).
This is unique across multiple indexed hie files (autoincremented id in sqlite).

Whereas `i` corresponds to index of a type within single hls file.

When I tried it duplicates started to be created again, not sure why.

Reply:
But we COULD maybe switch Int64 -> Int when looking up type indices here and then this would work?
https://github.com/wz1000/HieDb/blob/404b4fa5d597f9901afcaae0231519b1b5d04617/src/HieDb/Create.hs#L172-L173

Is there a need for such high type IDs there?

Reply:
You are correct about `i` not working. But I think sqlite indices are `Int64`, not much we can do about it.